lib/parse-message.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+module.exports = exports = {};
+
+exports.usernameCommand = function (message, socket) {
+  socket.username = message.split(' ').slice(1).join(' ').trim();
+  socket.write(`Your username is now ${socket.username}`);
+};
+
+exports.dmCommand = function (message, array, socket) {
+  let toUsername = message.split(' ')[1];
+  let content = message.split (' ').slice(2).join(' ').trim();
+  array.forEach(s => {
+    if (s.username === toUsername)
+      s.write(`${socket.username}${content}`);
+  });
+};
+
+exports.usersCommand = function(array, socket) {
+  array.forEach(users => {
+    socket.write(users.username + '\n');
```

Comment:
Get in the habit of using the new ES6 interpolation to concatenate strings because it'll be more practical in certain cases. You'll find it''ll much easier with some cases than keeping track of quotes and plus signs. E.g., the above would be:

```
`${users.username}\n`
```